src/org-manager/components/new-pbas-info/new-pbas-info.component.spec.ts

```diff
@@ -25,19 +51,43 @@ describe('NewPBAsInfoComponent', () => {
   beforeEach(() => {
     fixture = TestBed.createComponent(NewPBAsInfoComponent);
     component = fixture.componentInstance;
+    component.formGroup = new FormGroup({
+      PBA0101012: new FormControl('PBA0101012', Validators.required)
+    });
+    component.org = mockOrgData;
+    const map = new Map();
+    component.newPBAs = map.set('pba', 'PBA0101012');
     fixture.detectChanges();
   });
 
   it('should create', () => {
+    fixture.detectChanges();
+    component.formGroup = new FormGroup({
+      PBA0101012: new FormControl('PBA0101012', Validators.required)
+    });
     expect(component).toBeTruthy();
   });
 
   it('should call the submitForm output event emitter when onSubmit is called', () => {
     spyOn(component.submitForm, 'emit').and.callThrough();
+    const radioButton = fixture.debugElement.nativeElement.querySelector('#PBA0101012_approve');
+    radioButton.click();
+    fixture.detectChanges();
+    const submitButton = fixture.debugElement.nativeElement.querySelector('button[type="submit"]');
+    submitButton.click();
     component.onSubmitForm();
     expect(component.submitForm.emit).toHaveBeenCalled();
   });
 
+  it('should not submitForm if there is no radiobutton selected', () => {
+    spyOn(component.submitForm, 'emit').and.callThrough();
+    const submitButton = fixture.debugElement.nativeElement.querySelector('button[type="submit"]');
+    submitButton.click();
+    fixture.detectChanges();
+    component.onSubmitForm();
```

Comment:
Same as my previous comment - Is component.onSubmitForm needed now as you are calling the submitButton click event on line 85.

Reply:
done